File: Range Sum Query - Immutable/Range_Sum_Query_-_Immutable.java
class NumArray {
    private int[] prefix;

    public NumArray(int[] nums) {
        int n = nums.length;
        prefix = new int[n];
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            prefix[i] = sum;
        }
    }

    public int sumRange(int left, int right) {
        int ans = prefix[right];
        if (left > 0) {
            ans -= prefix[left - 1];
        }
        return ans;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: Range Sum Query - Immutable/Range_Sum_Query_-_Immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix = []
        sum = 0
        for num in nums:
            sum += num
            self.prefix.append(sum)
        

    def sumRange(self, left: int, right: int) -> int:
        ans = self.prefix[right]
        if left > 0:
            ans -= self.prefix[left - 1]
        return ans
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)